tests/test_dht.py

```diff
@@ -102,7 +102,7 @@ async def test_dht_get_visible_maddrs():
 
     dummy_endpoint = Multiaddr("/ip4/123.45.67.89/tcp/31337")
     p2p = await hivemind.p2p.P2P.create(announce_maddrs=[dummy_endpoint])
-    dht = hivemind.DHT(p2p, start=True)
+    dht = hivemind.DHT(start=True, p2p=await p2p.replicate(p2p.daemon_listen_maddr))
```

Comment:
This is the new canonic code to run DHT with a different P2P

Previous version had minor quirks: if P2P was destroyed in master (e.g. during fork, when it was dereferenced and subsequently garbage-collected), it would kill the daemon from under a running DHT